### ISSUE TITLE
is it needed for intel core i7 ?

### ISSUE THREAD

**Issue by Sandy4321** [fo40225/Anaconda-Windows-AMD#1]
is it needed for intel core i7 ?

**Comment by fo40225** [comment]
No, those whl is build for AMD.

Anaconda build numpy link against mkl, it provided maximum performance on Intel CPU already.

**Comment by RoyiAvital** [comment]
Could you explain what this library does?
Do you patch MKL to chose a good code path based on CPU features instead of the manufacturer?

**Comment by fo40225** [comment]
@RoyiAvital 
Those whl build with OpenMP instead of Intel TBB and link to Agner's MKL patch to prevent MKL check the CPUID is `GenuineIntel` or not.

**Comment by RoyiAvital** [comment]
Can you build MKL DLL like this?

I mean rebuilding something like `mkl_rt.dll`.

**Comment by fo40225** [comment]
@RoyiAvital 
No, the way to patch mkl on windows is static link to the mkl and replace the detect function.

Maybe you can replace the function in the `mkl_intel_thread.dll`, `mkl_avx2.dll`, `mkl_core.dll` or `mkl_sequential.dll`.

**Comment by RoyiAvital** [comment]
I see.

Problem is many languages use the `mkl_rt.dll`.
I wonder if there is a way to patch it or something like that.

**Comment by fo40225** [comment]
@RoyiAvital 
If you are using C++, checkout Agner's [asmlib](https://www.agner.org/optimize/#asmlib).

**Comment by RoyiAvital** [comment]
I know it. You seem to use its patch. I wonder how could one patch pre built DLL and not in compilation.

**Comment by fo40225** [comment]
@RoyiAvital 

You can write a wrapper dll that export all function in MKL but static link the mkl.lib and patch.

It should work.
